src/lib.rs

```diff
@@ -297,19 +324,31 @@ fn preprocess(content: &str, on_failure: OnFailure) -> MdbookResult<String> {
     let mut admonish_blocks = vec![];
 
     let events = Parser::new_ext(content, opts);
+
     for (e, span) in events.into_offset_iter() {
         if let Event::Start(Tag::CodeBlock(Fenced(info_string))) = e.clone() {
             let span_content = &content[span.start..span.end];
+
             let admonition = match parse_admonition(info_string.as_ref(), span_content, on_failure)
             {
                 Some(admonition) => admonition,
                 None => continue,
             };
-            let admonition = admonition?;
+
+            let mut admonition = admonition?;
+
+            // Handle any defaults given in context
+            admonition.attach_defaults(ctx)?;
```

Comment:
We can do default parsing once at the top, here you're doing it once per admonition.

I suppose this way we don't do it unless we have an admonition to process (you could make this lazy I suppose), but I assume if a user has installed the plugin they'll have a decent number of admonitions to handle..